analyzers/src/SonarAnalyzer.CSharp/Rules/CollectionsShouldImplementGenericInterface.cs

```diff
@@ -34,25 +34,28 @@ namespace SonarAnalyzer.Rules.CSharp
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     public sealed class CollectionsShouldImplementGenericInterface : SonarDiagnosticAnalyzer
     {
-        internal const string DiagnosticId = "S3909";
+        private const string DiagnosticId = "S3909";
         private const string MessageFormat = "Refactor this collection to implement '{0}'.";
 
         private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
 
-        private static readonly Dictionary<string, KnownType> NongenericToGenericMapping = new()
+        private static readonly Dictionary<KnownType, string> NonGenericToGenericMapping = new()
         {
-            { KnownType.System_Collections_ICollection.TypeName, KnownType.System_Collections_Generic_ICollection_T },
-            { KnownType.System_Collections_IList.TypeName, KnownType.System_Collections_Generic_IList_T },
-            { KnownType.System_Collections_IEnumerable.TypeName, KnownType.System_Collections_Generic_IEnumerable_T },
-            { KnownType.System_Collections_CollectionBase.TypeName, KnownType.System_Collections_ObjectModel_Collection_T },
+            { KnownType.System_Collections_ICollection, "System.Collections.Generic.ICollection<T>" },
```

Comment:
I've removed the old TypeName which had the full function signature and replaced it with the short version of the type. In the few cases where this was needed (~9), I hardcoded the values.